### ISSUE TITLE
Obligatory "Josh just wrote the docs and now has to complain about everything" issue

### ISSUE THREAD

**Issue by Yoshanuikabundi** [openforcefield/openff-units#27]
I'm putting this in an issue so that it doesn't get forgotten (by me if no-one else) when the docs PR is merged, and I'm putting it all in one issue to avoid spam. Here are my thoughts:

- [x] 1. `to_openmm` would make a great method for `Quantity`. This would save users an import when going in that direction. For bonus points, we could define it as a method on `Unit` as well and make the free function polymorphic over both
- [x] 2. having all the units be under `openff.units.unit` while the three big classes are under `openff.units.units` is confusing. ~If API breaks are OK,~ maybe we could ~rename `openff.units.units` to `openff.units._pint` and~ re-export `Quantity`, `Unit`, and `Measurement` from `openff.units` (either lazily or not) (Fixed in #28)
- [x] 3. It might be prudent to pin to a major version of Pint in each release. Our API is mostly inherited from theirs, so we can't really claim not to break things without a pin. In particular, our docs change a lot based on which version of Pint is installed in the RTD environment when its built. If we do this, it would be great to pick at least 0.18.0, as type hints were added to Quantity in that version.
- [ ] 4. It would be amazing to be able to specify the shape and units of a `Quantity[Array]` in type hints, does anyone know if that's possible? I know they're both part of the value and not the type but the two concepts are kinda fungible in Python. I vaguely remember that people somewhere somewhen were working on it.

**Comment by mattwthompson** [comment]
>  It would be amazing to be able to specify the shape and units of a Quantity[Array] in type hints, does anyone know if that's possible?

My best guess is that it's possible-but-difficult with PEP 646, but that's 3.11+ / I haven't looked for a backport.

**Comment by mattwthompson** [comment]
Oh, and for the record, I see four great suggestions and no complaints!

**Comment by mattwthompson** [comment]
After #34 I think 1-3 are now incorporated and 4 probably won't happen for a while. Would you be willing to copy+paste that into a new issue and close this one?

**Comment by Yoshanuikabundi** [comment]
@mattwthompson Too easy :)

Superseded by #35
